maestrowf/__init__.py

```diff
@@ -53,3 +60,372 @@ def emit(self, record):
 
 __version_info__ = ("1", "1", "9dev0")
 __version__ = '.'.join(__version_info__)
+
+
+# Register status layout types
+class StatusRendererFactory:
```

Comment:
There's two other factories that you could model one after -- almost wondering if it shouldn't just be made into a metaclass or something -- but those two base their keys on grabbing the classes of a specific base class. Another suggestion would be to add a method for getting the available renderers and then having it return a list of keys. That's useful in the `parser.add_argument(..., 'choices=StatusRendererFactory.keys( )')` optional so that the argument gets updated automatically.

Reply:
Ok, got this one and the other converted to use abstract base objects and got the auto-registration and cli population working.  This variant still keeps it in the __init__, but let me know if you'd prefer pushing the abstract base and concrete classes into submodules as with the script adapter setup.